File: src/AutoRest.SdkExplorer/Model/Azure/AzureServicePrincipal.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

#nullable disable

namespace AutoRest.SdkExplorer.Model.Azure
{
    public class AzureServicePrincipal
    {
        public string Name {get;set;}
        public string AppId  {get;set;}
        public string TenantId { get; set; }

        public AzureServicePrincipal(string name, string appId, string tenantId)
        {
            this.Name = name;
            this.AppId = appId;
            this.TenantId = tenantId;
        }
    }
}

File: src/AutoRest.SdkExplorer.TypeScript/src/Model/Azure/AzureServicePrincipal.ts
export class AzureServicePrincipal implements AutoRest.SdkExplorer.Interface.AzureServicePrincipal {
    name: string;
    appId: string;
    tenantId: string;

    constructor(data: AutoRest.SdkExplorer.Interface.AzureServicePrincipal) {
        this.name = data.name;
        this.appId = data.appId;
        this.tenantId = data.tenantId;
    }
}

